subsidy/views.py

```diff
@@ -54,8 +66,9 @@ def subsidy_delete(request, subsidy_id):
     return redirect("/subsidy/list")
 
 def subsidy_update(request, subsidy_id):
```

Comment:
Here you should add the `@login_required` decorator as you did on the create form.